pkg/connector/tickets.go

```diff
@@ -0,0 +1,449 @@
+package connector
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/url"
+	"path"
+	"strconv"
+	"time"
+
+	v2 "github.com/conductorone/baton-sdk/pb/c1/connector/v2"
+	"github.com/conductorone/baton-sdk/pkg/annotations"
+	"github.com/conductorone/baton-sdk/pkg/pagination"
+	sdkTicket "github.com/conductorone/baton-sdk/pkg/types/ticket"
+	"github.com/grpc-ecosystem/go-grpc-middleware/logging/zap/ctxzap"
+	"go.uber.org/zap"
+	"google.golang.org/protobuf/types/known/timestamppb"
+
+	jira "github.com/conductorone/go-jira/v2/cloud"
+)
+
+type TicketManager interface {
+	GetTicket(ctx context.Context, ticketId string) (*v2.Ticket, annotations.Annotations, error)
+	CreateTicket(ctx context.Context, ticket *v2.Ticket, schema *v2.TicketSchema) (*v2.Ticket, annotations.Annotations, error)
+	GetTicketSchema(ctx context.Context, schemaID string) (*v2.TicketSchema, annotations.Annotations, error)
+	ListTicketSchemas(ctx context.Context, pToken *pagination.Token) ([]*v2.TicketSchema, string, annotations.Annotations, error)
+}
+
+func ticketBuilder(j *Jira) TicketManager {
+	return j
+}
+
+func (j *Jira) ListTicketSchemas(ctx context.Context, p *pagination.Token) ([]*v2.TicketSchema, string, annotations.Annotations, error) {
+	var ret []*v2.TicketSchema
+
+	offset := 0
+	// get offset from page token if its not empty
+	if p != nil && p.Token != "" {
+		var err error
+		offset, err = strconv.Atoi(p.Token)
+		if err != nil {
+			return nil, "", nil, err
+		}
+	}
+
```

Comment:
Add request `https://swartkrans.atlassian.net/rest/api/3/statuses/search?expand=usages&statusCategory=DONE` paginate fully, probably just a single page, but we should paginate.